### PR TITLE
Allow greater or equal in storage provisioning check

### PR DESCRIPTION
**Pull request by awels**
Signed-off-by: Alexander Wels <awels@redhat.com>

#### What type of PR is this?

/kind cleanup
/kind failing-test

#### What this PR does / why we need it:
The current check enforces that a dynamic provisioner creates persistent volumes of capacity equals to the persistent volume
claim request size. However there are provisioners that will create persistent volumes with a capacity greater than the
request size (cinder comes to mind which increments in 1Gi increments, so if you request 0.5Gi, you get 1Gi). Also
provisioners that have shared storage should be reporting the total space not the request size (nfs/hostpath for instance).

All these will fail the provisioning check currently because the capacity is > than the request size. This modifies the check
to be capacity >= request size.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by awels** [comment]
I signed it

**Review by pohly — APPROVED** [review]
/ok-to-test
/lgtm

**Comment by pacoxu** [comment]
/hold
https://prow.k8s.io/pr-history/?org=kubernetes&repo=kubernetes&pr=107408
The pull-kubernetes-e2e-kind keeps failing. Would you take a look?

File: test/e2e/storage/testsuites/provisioning.go [hunk 1]
```diff
@@ -474,11 +474,11 @@ func (t StorageClassTest) checkProvisioning(client clientset.Interface, claim *v
 	// Check sizes
 	expectedCapacity := resource.MustParse(t.ExpectedSize)
 	pvCapacity := pv.Spec.Capacity[v1.ResourceName(v1.ResourceStorage)]
-	framework.ExpectEqual(pvCapacity.Value(), expectedCapacity.Value(), "pvCapacity is not equal to expectedCapacity")
+	gomega.Expect(pvCapacity.Value()).To(gomega.BeNumerically(">=", expectedCapacity.Value), "pvCapacity is not greater or equal to expectedCapacity")
```
Comment (pohly) [inline test/e2e/storage/testsuites/provisioning.go:477]:
```suggestion
	gomega.Expect(pvCapacity.Value()).To(gomega.BeNumerically(">=", expectedCapacity.Value()), "pvCapacity is not greater or equal to expectedCapacity")
```

File: test/e2e/storage/testsuites/provisioning.go [hunk 2]
```diff
@@ -474,11 +474,11 @@ func (t StorageClassTest) checkProvisioning(client clientset.Interface, claim *v
...
 
 	requestedCapacity := resource.MustParse(t.ClaimSize)
 	claimCapacity := claim.Spec.Resources.Requests[v1.ResourceName(v1.ResourceStorage)]
-	framework.ExpectEqual(claimCapacity.Value(), requestedCapacity.Value(), "claimCapacity is not equal to requestedCapacity")
+	gomega.Expect(claimCapacity.Value()).To(gomega.BeNumerically(">=", requestedCapacity.Value), "claimCapacity is not greater or equal to requestedCapacity")
```
Comment (pohly) [inline test/e2e/storage/testsuites/provisioning.go:481]:
```suggestion
	gomega.Expect(claimCapacity.Value()).To(gomega.BeNumerically(">=", requestedCapacity.Value()), "claimCapacity is not greater or equal to requestedCapacity")
```

**Review by pohly — CHANGES_REQUESTED** [review]
Some brackets got lost... :sweat_smile:

**Comment by awels** [comment]
Weird, it compiled for me locally.

**Comment by awels** [comment]
So I just did a naive go build *file* but obviously that is not correct. If I make more changes to some of these tests. What would be the right command to compile it all properly so I don't have these silly brackets are missing failures.

**Review by pohly — APPROVED** [review]
/lgtm

**Comment by pohly** [comment]
> So I just did a naive go build file but obviously that is not correct. If I make more changes to some of these tests. What would be the right command to compile it all properly so I don't have these silly brackets are missing failures.

The gomega parameters are untyped (`interface{}`), therefore compilation alone can't catch this. You would have to run the test (`go test -v ./test/e2e -args -ginkgo.focus=<test name>`).

/hold cancel

**Pull request merged**